src/rules/noVarKeywordRule.ts

```diff
@@ -37,46 +38,50 @@ export class Rule extends Lint.Rules.AbstractRule {
+    const { sourceFile } = ctx;
+    return ts.forEachChild(ctx.sourceFile, function cb(node: ts.Node): void {
+        switch (node.kind) {
+            case ts.SyntaxKind.VariableStatement: {
+                const vs = node as ts.VariableStatement;
+                if (!Lint.isBlockScopedVariableDeclarationList(vs.declarationList) && !isGlobalVarDeclaration(vs)) {
+                    fail(vs.declarationList);
+                }
+                break;
+            }
 
-    public visitForStatement(node: ts.ForStatement) {
-        this.handleInitializerNode(node.initializer);
-        super.visitForStatement(node);
-    }
+            case ts.SyntaxKind.ForStatement:
+            case ts.SyntaxKind.ForInStatement:
+            case ts.SyntaxKind.ForOfStatement: {
+                const { initializer } = node as ts.ForStatement | ts.ForInStatement | ts.ForOfStatement;
+                if (initializer && isVariableDeclarationList(initializer) && !Lint.isBlockScopedVariableDeclarationList(initializer)) {
+                    fail(initializer);
+                }
+                break;
+            }
+        }
 
-    public visitForInStatement(node: ts.ForInStatement) {
-        this.handleInitializerNode(node.initializer);
-        super.visitForInStatement(node);
-    }
+        return ts.forEachChild(node, cb);
+    });
 
-    public visitForOfStatement(node: ts.ForOfStatement) {
-        this.handleInitializerNode(node.initializer);
-        super.visitForOfStatement(node);
+    function fail(node: ts.Node): void {
+        // Don't apply fix in a declaration file, because may have meant 'const'.
+        const fix = sourceFile.isDeclarationFile ? undefined : new Lint.Replacement(node.getStart(), "var".length, "let");
+        ctx.addFailureAtNode(Lint.childOfKind(node, ts.SyntaxKind.VarKeyword)!, Rule.FAILURE_STRING, fix);
     }
+}
 
-    private handleInitializerNode(node: ts.VariableDeclarationList | ts.Expression | undefined) {
-        if (node && node.kind === ts.SyntaxKind.VariableDeclarationList &&
-                !(Lint.isNodeFlagSet(node, ts.NodeFlags.Let) || Lint.isNodeFlagSet(node, ts.NodeFlags.Const))) {
-            this.reportFailure(node);
-        }
-    }
+// Allow `declare var x: number;` or `declare global { var x: number; }`
+function isGlobalVarDeclaration(node: ts.VariableStatement): boolean {
+    const parent = node.parent!;
+    return Lint.hasModifier(node.modifiers, ts.SyntaxKind.DeclareKeyword)
+        || parent.kind === ts.SyntaxKind.ModuleBlock && isDeclareGlobal(parent.parent!);
+}
 
-    private reportFailure(node: ts.Node) {
-        const nodeStart = node.getStart(this.getSourceFile());
-        this.addFailureAt(nodeStart, "var".length, Rule.FAILURE_STRING,
-            this.createReplacement(nodeStart, "var".length, "let"));
-    }
+function isDeclareGlobal(node: ts.Node): boolean {
+    return isModuleDeclaration(node) && node.name.kind === ts.SyntaxKind.Identifier && node.name.text === "global";
```

Comment:
you could even simplify this further:
```ts
return (node.flags & ts.NodeFlags.GlobalAugmentation) !== 0;
```
In addition you avoid false positived for `namespace foo.global {}` and `namespace foo { export namespace global {} }` (those are also good candidates for a test)

You can even consider inlining this condition.